autogpt_platform/backend/backend/server/rest_api.py

```diff
@@ -539,7 +542,9 @@ async def stop_graph_run(
                 404, detail=f"Agent execution #{graph_exec_id} not found"
             )
 
-        self.execution_manager_client.cancel_execution(graph_exec_id)
+        await asyncio.to_thread(
+            lambda: self.execution_manager_client.cancel_execution(graph_exec_id)
+        )
```

Comment:
pyro call inside the async function need to do this, to avoid blocking the event loop in case the function call is slow.